src/solvers/flattening/boolbv_add_sub.cpp

```diff
@@ -91,7 +102,8 @@ bvt boolbvt::convert_add_sub(const exprt &expr)
 
         for(std::size_t j=0; j<tmp_op.size(); j++)
         {
-          assert(i*sub_width+j<op.size());
+          INVARIANT(
+            i * sub_width + j < op.size(), "bit index shall be within bounds");
```

Comment:
I don't think we need to check this `tmp_op.size()`-times. We can just check this once before entering the loop.

Reply:
The invariant is actually contained in a nested loop. It seems that the intention is also to check that we didn't mess up the index computation (e.g. by confusing `i` and `j`). How about we keep the invariant at this place and introduce a new local variable `index` which is set to `i * sub_width + j` and then use this variable in both the invariant and the array access?

Reply:
Ok, that would make sense.